### ISSUE TITLE
drives redirection error

### ISSUE THREAD

**Issue by mightyraven** [neutrinolabs/xrdp#761]
hi all.
When connecting to a remote server first time "drive redirection" works  (directory "thinclient_drives" created)
Complete the session.
Сonnecting the second time - "drive redirection" doesn't work (directory thinclient_drives becomes red with name "?thinclient_drives")

**Comment by mightyraven** [comment]
I have the same problem as here: https://bugs.debian.org/cgi-bin/bugreport.cgi?bug=831873

**Comment by mightyraven** [comment]
The problem occurs when connecting from Windows 7 (mstsc). When I connect from debian (xfreerdp) everything works fine.

**Comment by metalefty** [comment]
Can you show me client's redirection setting? Not only drives, clipboard, printers, smartcard.

**Comment by mightyraven** [comment]
client's redirection setting:
smartcard - no
ports - no
drive - yes
clipboard - yes
sound - yes

**Comment by metalefty** [comment]
and printers?

**Comment by mightyraven** [comment]
more information: 
when connecting from Windows 7 (mstsc), I have created the file: .xorgxrdp.10.log .xorgxrdp.11.log .xorgxrdp.12.log .xorgxrdp.13.log, and so on. 
When I connect from debian (xfreerdp), the file is always the same - .xorgxrdp.10.log

**Comment by mightyraven** [comment]
printers - no!!!

**Comment by metalefty** [comment]
I don't see you issue your issue is as same as Debian bug 831873. Maybe wrong url?

**Comment by mightyraven** [comment]
Url is correct. I checked. Try again click on the link https://bugs.debian.org/cgi-bin/bugreport.cgi?bug=831873

**Comment by mightyraven** [comment]
An error occurs even within one session. For example, I'm connected by a simple user - everything works (user_mc.png 
![user_mc](https://user-images.githubusercontent.com/25559627/27373675-6d8e3e02-5693-11e7-8f27-96601baa39b1.png)
). Then, if I become root (using the SU utility), the directory "thinclient_drives" does not work (root_mc.png 
![root_mc](https://user-images.githubusercontent.com/25559627/27373725-994ec9c6-5693-11e7-97f8-44ee65008ab3.png)
).

**Comment by zertrin** [comment]
@mightyraven concerning the issue when becoming root, this is not a bug, but the way FUSE works by default;

See https://unix.stackexchange.com/questions/17402/why-does-root-get-permission-denied-when-accessing-fuse-directory

**Comment by metalefty** [comment]
It is normal FUSE behaviour. The mountpoint is only available for the user who mounted the mountpoint.

**Comment by metalefty** [comment]
What is your real problem?

becoming root?
${HOME}/.thinclient_drives which has permissions 000 belongs to root (Debian bug 831873)?
or else? 

I don't see you're talking about a single problem.

**Comment by mightyraven** [comment]
real problem is #780 (umount)

**Comment by metalefty** [comment]
OK, I understood. I'll investigate the unmount issue.

**Comment by mightyraven** [comment]
> OK, I understood. I'll investigate the unmount issue.

Thank you. (If you need log files, I'll post). Once again many thanks for the xrdp-project

**Comment by mightyraven** [comment]
For more information on the issue: 

When I log in under root on the server (locally, without the use of xrdp), and then connect remotely under the user using xrdp, problems with drive redirection does not occur. Disconnect, then reconnect. Everything is working fine. 
But as soon as I log off root (locally, without the use of xrdp), disconnect and reconnect the drive redirection is broken.

**Comment by DRN88** [comment]
It looks like it is still broken.

```bash
[root@lxcentos7test01 ~]# cat /etc/redhat-release 
CentOS Linux release 7.3.1611 (Core) 
[root@lxcentos7test01 ~]# rpm -qa | grep xrdp
xorgxrdp-0.2.3-1.el7.x86_64
xrdp-0.9.2-12.el7.x86_64
```

Meanwhile I am trying to find a `mate-session` end hook or something similar where I can unmount it manually after logout. Any ideas welcome. Thanks.

**Comment by DRN88** [comment]
Very simple workaround is the following:
`fusermount -u "$HOME/redirected_drives"`

My custom `startwm.sh` looks like this:

```bash
[root@lxcentos7test01 ~]# cat /usr/libexec/xrdp/startwm-mate-session.sh 
#!/bin/sh
# Ansible managed
mate-session
fusermount -u "$HOME/redirected_drives"
exit 1
```

Set drive folder name manually:
```bash
[root@lxcentos7test01 ~]# grep FuseMountName /etc/xrdp/sesman.ini 
FuseMountName=redirected_drives
```

**Comment by schiffermtq** [comment]
Has there been any progress on this issue? I'm still seeing this with xrdp 0.9.6 (on Ubuntu 16.04).

**Comment by Gerhard-Vienna** [comment]
On RedHat 7 with KDE as your desktop (xrdp package 1:0.9.6-4.el7):

Create a script:
~/.kde/shutdown/end_kde 
 
containing just this line:
fusermount -u ~/thinclient_drives
an give it execute permissions:
chmod a+x ~/.kde/shutdown/end_kde 

Done...

If things are really messed up you might need to do (as root):
touch /forcefsck
reboot

**Comment by metalefty** [comment]
Fixed by #1140.
